src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlInternalConnection.cs

```diff
@@ -186,6 +207,20 @@ internal bool IsGlobalTransactionsEnabledForServer
             }
         }
 
+#if NETFRAMEWORK
+        internal bool IsAzureSQLConnection
```

Comment:
There is code in netfx that uses it to adjust behavior:
```
// If not a connection to Azure SQL, Readonly with FailoverPartner is not supported
```
I don't know why the same code is not in netcore. I suspect it should be, but it would require a little investigation.